src/Backups/BackupIO.cpp

```diff
@@ -22,6 +23,11 @@ void IBackupReader::copyFileToDisk(const String & file_name, size_t size, DiskPt
     write_buffer->finalize();
 }
 
+IBackupWriter::IBackupWriter(const ContextPtr & context_)
+    : read_settings(context_->getBackupReadSettings())
+    , has_throttling(static_cast<bool>(context_->getBackupsThrottler()))
```

Comment:
Why don't you get your throttler from `read_settings`?
(i.e. from `context_->getBackupReadSettings()`)

Reply:
Yes, but this is more explicit to me. I don't have strong opinion here, let me know which one will be more feasible for upstream.

Reply:
Ok, it doesn't matter.